plugin/builtin/attach/attach_plugin.go

```diff
@@ -64,26 +82,30 @@ func (self *AttachPlugin) GetPanelConfig() (*plugin.PanelConfig, error) {
 					if context.Task == nil {
 						return nil, nil
 					}
+					var err error
 					taskId := context.Task.Id
+					t := context.Task
 					if context.Task.OldTaskId != "" {
 						taskId = context.Task.OldTaskId
+						t, err = task.FindOneId(taskId)
+						if err != nil {
+							return nil, errors.Wrap(err, "error retrieving task")
+						}
 					}
 
-					artifactEntry, err := artifact.FindOne(artifact.ByTaskIdAndExecution(taskId, context.Task.Execution))
+					files, err := getAllArtifacts(taskId, context.Task.Execution)
 					if err != nil {
-						return nil, errors.Wrap(err, "error finding artifact files for task")
+						return nil, err
 					}
-					if artifactEntry == nil {
 
-						artifactEntry, err = artifact.FindOne(artifact.ByTaskIdWithoutExecution(taskId))
+					for _, execTask := range t.ExecutionTasks {
+						execTaskFiles, err := getAllArtifacts(execTask, context.Task.Execution)
```

Comment:
feels like this could/should be a $in query rather than doing a bunch of finds in a loop.